src/System/IO/Extra.hs

```diff
@@ -109,19 +111,19 @@ writeFileBinary file x = withBinaryFile file WriteMode $ \h -> hPutStr h x
 --
 -- > captureOutput (print 1) == return ("1\n",())
 captureOutput :: IO a -> IO (String, a)
-captureOutput act = withTempFile $ \file -> do
+captureOutput act = withTempFile $ \file ->
     withFile file ReadWriteMode $ \h -> do
         res <- clone stdout h $ clone stderr h $ do
             hClose h
             act
         out <- readFile' file
         return (out, res)
     where
-        clone out h act = do
+        clone out h act' = do
```

Comment:
I realise the shadowing upsets `-Wall` but it's 100% deliberate. By shadowing `act` I ensure I can't accidentally call the original `act` in this function, and thus the most likely bug goes away.